core/src/test/java/io/jenkins/pluginhealth/scoring/probes/FailingBuildProbeTest.java

```diff
@@ -0,0 +1,33 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+
+import org.junit.jupiter.api.Test;
+
+public class FailingBuildProbeTest extends AbstractProbeTest<FailingBuildProbe> {
+
+    @Override
+    FailingBuildProbe getSpy() {
+         return spy(FailingBuildProbe.class);
+    }
+
+    @Test
+    void shouldCorrectlyDetectMissingJenkinsfile() throws IOException {
+        final FailingBuildProbe probe = getSpy();
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final Path repo = Files.createTempDirectory("foo");
+        when(ctx.getScmRepository()).thenReturn(Optional.of(repo));
+        assertThat(probe.apply(plugin, ctx)).withFailMessage("No JenkinsFile found");
+    }
```

Comment:
I think this is already taking care off by the Jenkinsfile probe.

What I think would be beneficial for this pull request is to validate the behaviour when we cannot get the run of the latest commit of the default branch, to make sure which "run" you are getting with the `iterator().next()`, if it's possible for the conclusion to be `null`.

Reply:
Thanks @alecharp Is the above logic to check the status of the default branch in the probe is not correct ?

Reply:
Do you mean https://github.com/jenkins-infra/plugin-health-scoring/pull/422/files#diff-ccfb769cc30831bdce2acc485da9fb8ffa7d11a932657c1528b75f02e58e7f04R60-R71? Probably.